src/toolbox.ts

```diff
@@ -1999,56 +2017,36 @@ export class FilterPointDistanceFromRow extends ToolboxItem {
         // Get this component's config from ulabel's config
         this.config = this.ulabel.config.distance_filter_toolbox_item
 
-        // Create a set of defaults for every config value
-        const default_values = {
-            "name": <string> "Filter Distance From Row",
-            "component_name": <string> "fitler_distance_from_row",
-            "filter_min": <number> 0,
-            "filter_max": <number> 100,
-            "default_values": <AnnotationClassDistanceData> {"single": 50},
-            "step_value": <number> 1,
-            "multi_class_mode": <boolean> false,
-            "filter_on_load": <boolean> true,
-            "show_options": <boolean> true,
-            "toggle_overlay_keybind": <string> "p"
-        }
-
-        // Loop through every key value pair in the config
-        for (const [key, value] of Object.entries(this.config)) {
-            // If the passed in value's type !== the default value's type then use the default value
-            if (typeof value !== typeof default_values[key]) {
-                this.config[key] = default_values[key]
+        // For each key missing from the config, set the default value
+        for (const key in DEFAULT_FILTER_DISTANCE_CONFIG) {
+            if (!this.config.hasOwnProperty(key)) {
+                this.config[key] = DEFAULT_FILTER_DISTANCE_CONFIG[key]
```

Comment:
Is it a big lift to just make this a class with defaults? Or do we foresee multiple sets of defaults for this?

Reply:
I reworked this because before there were defaults in two places: `configuration.ts` and `toolbox.ts`. Is your suggestion that the defaults should just be in `toolbox.ts`? I don't feel strongly either way. I went with `configuration.ts` since that's where all the other constructor arg defaults are gathered, but could be convinced that we should move it here.

Reply:
Well, on account of `toolbox.ts` being very big, and it being logical to keep all configurable settings in one place, this is fine.

Copying from a default like this is the part that feels weird - if there was a generic function for applying anything that matched the config type, that then got passed `DEFAULT_FILTER_DISTANCE_CONFIG`, I'd feel a little better about it. Getting a little bit nit-picky though, so we can also punt on this if you're not concerned.

Reply:
Yeah thinking about it further, punt for now. It will come up if the `ToolboxItem` hierarchy ever gets reworked/separated into their own files.